src/api/offers.js

```diff
@@ -50,89 +54,100 @@ export default ({offerService}: { offerService: OfferService }): Router => {
     body('duration').isInt().toInt().custom((value: number): boolean => [3, 7, 14, 30].includes(value)),
     body('formData').exists(),
     body('agreedToDataProtection').isBoolean().toBoolean().custom((value: boolean): boolean => value),
-    validateMiddleware,
+    validateMiddleware(errorService),
     async (request: $Request, response: $Response): Promise<void> => {
       const {email, formData, duration} = matchedData(request)
       try {
         const token = await offerService.createOffer(request.city, email, formData, duration)
+
         if (develop) {
           response.status(HttpStatus.CREATED).json(token)
         } else {
           response.status(HttpStatus.CREATED).end()
         }
       } catch (e) {
-        console.error(e)
-        response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
+        let errorResponse
+        // Check if the error is a mongoose ValidationError
+        if (e.name && e.name === 'ValidationError') {
+          errorResponse = errorService.createValidationFailedErrorResponse(e)
+          response.status(HttpStatus.BAD_REQUEST).json(errorResponse)
+        } else {
+          errorResponse = errorService.createInternalServerErrorResponse(e)
+          response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(errorResponse)
+        }
       }
     }
   )
 
   router.post(`/:token/confirm`,
     param('token').isHexadecimal().isLength(TOKEN_LENGTH),
-    validateMiddleware,
+    validateMiddleware(errorService),
     async (request: $Request, response: $Response): Promise<void> => {
       try {
         const {token} = matchedData(request)
         const offer = await offerService.getOfferByToken(token)
 
-        if (!offer) {
-          response.status(HttpStatus.NOT_FOUND).json('No such offer')
-        } else if (offer.expirationDate <= Date.now() || offer.deleted) {
-          response.status(HttpStatus.GONE).json('Offer not available')
+        if (!offer || offer.deleted) {
+          const errorResponse = errorService.createOfferNotFoundErrorResponse(token)
+          response.status(HttpStatus.NOT_FOUND).json(errorResponse)
+        } else if (offer.expirationDate <= Date.now()) {
+          const errorResponse = errorService.createOfferExpiredErrorResponse(token)
+          response.status(HttpStatus.GONE).json(errorResponse)
         } else {
           await offerService.confirmOffer(offer, token)
           response.status(HttpStatus.OK).end()
         }
       } catch (e) {
-        console.error(e)
-        response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
+        const errorResponse = errorService.createInternalServerErrorResponse(e)
+        response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(errorResponse)
       }
```

Comment:
Currently you surround every api method with a try, catch block.
You could also add this as middleware within the api/index.js:
```js
const catchInternalErrors = (request, response, next) => {
  try {
    next()
  } catch (e) {
    // todo: errorService should do stuff
  }
}
```
and then somewhere (before all other calls to `api`) you would put 
```js
api.use(catchInternalErrors)
```

That way, we don't need to set a a try, catch block everywhere.

http://expressjs.com/de/guide/using-middleware.html